packages/server/src/fhir/operations/agentpush.ts

```diff
@@ -118,47 +107,26 @@ async function pushToAgent(
     body: params.body,
   };
 
-  // If not waiting for a response, publish and return
-  if (!params.waitForResponse) {
-    await publishMessage(agent, message);
-    sendOperationResponse(allOk);
+  // Publish the message to the agent channel
+  const [outcome, response] = await publishAgentMessage<AgentTransmitResponse>(
+    agent,
+    message,
+    params.waitForResponse ? { waitForResponse: true, timeout: waitTimeout } : undefined
+  );
+
+  if (!response) {
+    sendOperationResponse(outcome);
```

Comment:
Non-blocking, future work: It might be nice to revisit whether this could have been done with `async` rather than promises and callbacks.